### PR TITLE
Methods for updating network sections

### PR DESCRIPTION
**Pull request by inercia**
This PR adds a method for updating network definition sections.

### CODE REVIEW

File: network.go [hunk 1]
```diff
@@ -138,6 +170,44 @@ func (n *VirNetwork) GetXMLDesc(flags uint32) (string, error) {
...
+	}
+	if len(mac) > 0 {
+		b.WriteString(fmt.Sprintf(" mac=\"%s\"", mac))
+	}
+	if len(name) > 0 {
+		b.WriteString(fmt.Sprintf(" name=\"%s\"", name))
+	}
+	b.WriteString(" />")
+	return b.String()
+}
```
Comment (flavio) [inline network.go]: What about using XML marshal for that? That would let me sleep tight at night :)

I can take care of that of course

Comment (flavio) [inline network.go]: I'm on that

Comment (inercia) [inline network.go]: You will end up with more lines of code with the XML marshalling thing, but it is ok if you want...

File: network.go [hunk 2]
```diff
@@ -138,6 +170,44 @@ func (n *VirNetwork) GetXMLDesc(flags uint32) (string, error) {
...
+		C.VIR_NETWORK_UPDATE_COMMAND_ADD_LAST, C.VIR_NETWORK_SECTION_IP_DHCP_HOST)
+}
+
+// Removes a static host from the network
+func (n *VirNetwork) RemoveHost(ip, mac, name string) error {
+	return n.UpdateXMLDesc(getHostXMLDesc(ip, mac, name),
+		C.VIR_NETWORK_UPDATE_COMMAND_DELETE, C.VIR_NETWORK_SECTION_IP_DHCP_HOST)
+}
+
 func (n *VirNetwork) Undefine() error {
```
Comment (dmacvicar) [inline network.go:181]: I don't think I will be able to upstream all of this. libvirt go are bindings to the library and you are adding functions that are not in the libvirt public API.

Comment (flavio) [inline network.go:181]: Good point. It looks like we should move these functions into the terraform code. Am I right?

Comment (inercia) [inline network.go:181]: Ok, I'll move this to the terraform side of things...

File: network.go (hunk 1 above)
Comment (inercia) [inline network.go]: Moved to the Terraform provider

File: network.go (hunk 2 above)
Comment (inercia) [inline network.go:181]: Done.

**Comment by flavio** [comment]
LGTM

**Pull request merged**